src/multi_vector_simulator/C1_verification.py

```diff
@@ -766,3 +770,87 @@ def check_for_sufficient_assets_on_busses(dict_values):
                 f"so that the bus is not a dead end should be two, excluding the excess sink. "
                 f"These are the connected assets: {asset_string}"
             )
+
+
+def check_energy_system_can_fulfill_max_demand(dict_values):
+    r"""
+    Helps to do oemof-solph termination debugging: Logs a logging.warning message if the aggregated installed capacity and maximum capacity (if applicable)
+    of all conversion, generation and storage assets connected to one bus is smaller than the maximum demand.
+
+   The check is applied to each bus of the energy system. Check passes when the potential peak supply is
+   larger then or equal to the peak demand on the bus, or if the maximum capacity of an asset is set to
+   None when optimizing.
+
+    Parameters
+    ----------
+    dict_values : dict
+        Contains all input data of the simulation.
+
+    Returns
+    -------
+    Indirectly, logs a logging.warning message if the installed and maximum capacities of
+    conversion/generation/storage assets are less than the maximum demand, for each bus.
+
+    Notes
+    -----
+    Tested with:
+    - test_check_energy_system_can_fulfill_max_demand_sufficient_capacities()
+    - test_check_energy_system_can_fulfill_max_demand_no_maximum_capacity()
+    - test_check_energy_system_can_fulfill_max_demand_insufficient_capacities()
+    - test_check_energy_system_can_fulfill_max_demand_with_storage()
+    """
+    for bus in dict_values[ENERGY_BUSSES]:
+        pass_check = False
+        opt_cap_storage = False
+        peak_demand = 0
+        peak_generation = 0
+        for item in dict_values[ENERGY_BUSSES][bus][ASSET_DICT]:
+            # filters out excess energy sinks, leaving only actual demand profiles
+            if item in dict_values[ENERGY_CONSUMPTION]:
+                if DISPATCHABILITY in dict_values[ENERGY_CONSUMPTION][item]:
+                    if (
+                        dict_values[ENERGY_CONSUMPTION][item][DISPATCHABILITY][VALUE]
+                        is False
+                    ):
+                        peak_demand += dict_values[ENERGY_CONSUMPTION][item][
+                            TIMESERIES_PEAK
+                        ][VALUE]
+            for group in [ENERGY_CONVERSION, ENERGY_PRODUCTION]:
+                if item in dict_values[group]:
+                    peak_generation += dict_values[group][item][INSTALLED_CAP][VALUE]
+                    if dict_values[group][item][OPTIMIZE_CAP][VALUE] is True:
+                        if dict_values[group][item][MAXIMUM_CAP][VALUE] is None:
+                            pass_check = True
+                        else:
+                            peak_generation += dict_values[group][item][MAXIMUM_CAP][
+                                VALUE
+                            ]
```

Comment:
...actually I also noticed right now that we did a little mistake with the EnergyConversion asset as well - the conversion assets are uni-directional, meaning that not all of them can actually supply the demand. We should take that into account. Therefore, to fix both things, I propose:
```suggestion
            # Only add capacity of conversion assets that can contribute to supply
             if item in dict_values[ENERGY_CONVERSION] and  dict_values[ENERGY_CONVERSION] [item][OUTFLOW_DIRECTION]==bus:
                    peak_generation += dict_values[ENERGY_CONVERSION][item][INSTALLED_CAP][VALUE]
                    if dict_values[ENERGY_CONVERSION][item][OPTIMIZE_CAP][VALUE] is True:
                        if dict_values[ENERGY_CONVERSION][item][MAXIMUM_CAP][VALUE] is None:
                            pass_check = True
                        else:
                            peak_generation += dict_values[ENERGY_CONVERSION][item][MAXIMUM_CAP][
                                VALUE
                            ]
                  if item in dict_values[ENERGY_PRODUCTION]:
                    # Effective generation of asset dependent on the peak of timeseries
                    peak_generation += dict_values[ENERGY_PRODUCTION][item][INSTALLED_CAP][VALUE]*dict_values[ENERGY_PRODUCTION][item][TIMESERIES_PEAK][VALUE]
                    if dict_values[ENERGY_PRODUCTION][item][OPTIMIZE_CAP][VALUE] is True:
                        if dict_values[ENERGY_PRODUCTION][item][MAXIMUM_CAP][VALUE] is None:
                            pass_check = True
                        else:
                           # Effective generation of asset dependent on the peak of timeseries
                            peak_generation += dict_values[ENERGY_PRODUCTION][item][MAXIMUM_CAP][
                                VALUE
                            ]*dict_values[ENERGY_PRODUCTION][item][TIMESERIES_PEAK][VALUE]
```
(Indentation will be off)
This still needs to be metioned in #829, as this decision will effect how we deal with the production possible from `InstalledCap` and `MaximumCap`.